src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509SecurityTokenProvider.cs

```diff
@@ -34,9 +34,9 @@ internal X509SecurityTokenProvider(X509Certificate2 certificate, bool clone)
             }
         }
 
-        protected override async Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
+        protected override Task<SecurityToken> GetTokenCoreAsync(CancellationToken cancellationToken)
         {
-            return await Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
+            return Task.FromResult<SecurityToken>(new X509SecurityToken(certificate: _certificate, clone: _clone, disposable:_clone));
```

Comment:
You're right, I missed that. They really should have made Task<T> covariant, that catches me from time to time. So it's good as it is.